src/cljs/rems/catalogue.cljs

```diff
@@ -166,18 +166,24 @@
      [document-title (text :t.catalogue/catalogue)]
      [flash-message/component :top]
      (text :t.catalogue/intro)
-     (if (or @(rf/subscribe [::full-catalogue :fetching?])
-             @(rf/subscribe [::full-catalogue-tree :fetching?])
-             @(rf/subscribe [::draft-applications :fetching?]))
-       [spinner/big]
-       [:div
-        (when @(rf/subscribe [:logged-in])
-          [:<>
-           [draft-application-list]
-           (when (:enable-cart config)
-             [cart/cart-list-container])
-           [:h2 (text :t.catalogue/apply-resources)]])
-        (when (:enable-catalogue-tree config)
-          [catalogue-tree])
-        (when (:enable-catalogue-table config)
-          [catalogue-table])])]))
+     [:div
+      (when @(rf/subscribe [:logged-in])
+        [:<>
+         [draft-application-list]
+
+         (when (:enable-cart config)
+           (when-not (or @(rf/subscribe [::full-catalogue :fetching?])
+                         @(rf/subscribe [::full-catalogue-tree :fetching?]))
+             [cart/cart-list-container]))
+
+         [:h2 (text :t.catalogue/apply-resources)]])
+
+      (when (:enable-catalogue-tree config)
+        (if @(rf/subscribe [::full-catalogue :fetching?])
+          [spinner/big]
+          [catalogue-tree]))
+
+      (when (:enable-catalogue-table config)
+        (if @(rf/subscribe [::full-catalogue-tree :fetching?])
+          [spinner/big]
+          [catalogue-table]))]]))
```

Comment:
faster rendering is :+1:

with larger data, even though catalogue now loads a lot faster, there seems to be a rendering gap: if catalogue-table is enabled, it widens the layout once filled with data.

something that could be looked into as later improvement, maybe with bootstrap update

Reply:
Currently all the pages suffer from the same issue of layout shift. The root cause is that I removed the fixed / max width some time ago. This seemed to be the only way to have the tables in the admin side to fill only as much space as they need and be centered.